src/agora/dozf.cc

```diff
@@ -391,39 +412,35 @@ void DoZF::ZfFreqOrthogonal(size_t tag) {
   duration_stat_->task_duration_[1] += start_tsc2 - start_tsc1;
 
   if (cfg_->Frame().NumDLSyms() > 0) {
-    arma::cx_fvec calib_vec(
-        reinterpret_cast<arma::cx_float*>(calib_gather_buffer_),
-        cfg_->BfAntNum(), false);
     size_t frame_cal_slot = kFrameWnd - 1;
-    size_t frame_cal_slot_prev = kFrameWnd - 1;
-    if (cfg_->Frame().IsRecCalEnabled() && (frame_id >= TX_FRAME_DELTA)) {
-      size_t frame_grp_id = (frame_id - TX_FRAME_DELTA) / cfg_->AntGroupNum();
+    size_t frame_cal_slot_prev = kFrameWnd - 2;
+    if (cfg_->Frame().IsRecCalEnabled()) {
+      const size_t frames_to_complete = cfg_->RecipCalFrameCnt();
+      const size_t current_cal_index = frame_id / frames_to_complete;
```

Comment:
Should we make these two lines a function too since it is repeated throughout?

Reply:
Reworked